### DISCUSSION TITLE
Version 3.6 not respect margins and size paper

### DISCUSSION THREAD

**Discussion by dominiosc** [sumatrapdfreader/sumatrapdf#3830] [General]
I don't know if _"The update fixes some PDF documents."_ result to broke others.
I use sumatra to print labels,  when update at least to 3.6.15810 the size of paper is changed
![from-3 6 15810](https://github.com/sumatrapdfreader/sumatrapdf/assets/63964184/9f377cf0-3853-403f-b7af-25ac219e427b)

In previous version, 3.5.2 work correctly
![previous-3 6 15810](https://github.com/sumatrapdfreader/sumatrapdf/assets/63964184/9b558dd2-f503-4f96-8f21-c12a1a27bdbf)

Also, when cut blank margins, sumatra respect top and bottom margins, but not show correctly left and right, in previous versions work correctly, and other programs too.
![sum](https://github.com/sumatrapdfreader/sumatrapdf/assets/63964184/aa810d87-91b0-46e2-8390-9b2801f58df9)

Please check where is the problem to fix it.
Thanks

**Comment by GitHubRulesOK** [comment]
Sounds like a crop box issue and use of default paper
but without samples please nothing to fix
SumatraPDF does not pad PDF
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/5f794de4-3a8d-45ad-8e6d-0a6bcf0d1689)

**Comment by dominiosc** [comment]
The problem not is view only, when print the label use more than one label and off center.

I share example files of first two cases, that size is more small, you have seven days to expire the link.
https://www.dropbox.com/t/b1QSAXYlwFnu4mmS

The third case i think is the same problem, the pdf file is more bigger, but is necessary I send too.

Thanks

**Reply by GitHubRulesOK** [reply]
sorry the answer below is wrong but it covers all issues except

 you are right latest version SumatraPDF is not handling media size the same as before 

OK there appears to be a problem so raised issue https://github.com/sumatrapdfreader/sumatrapdf/issues/3834

**Comment by GitHubRulesOK** [comment]
Thanks for Samples
My initial observations are these

1. ) as expected irregular sizes so will not work well with print as image by defaults without some alterations to windows FORMS or very specific control at the printer
2. ) include monochrome labels so again not suited to colour printing where each black pixel needs to be spread into CMYK at the print head which for label printer will just be K thus the result usually fuzzy where the CMY dots would be this usually produces ragged edges to barcodes that make them less or non readable.
3. ) the output print driver needs to align colours to black and that often means a dedicated mono print driver , certainly NOT MS Print to PDF.

Overall SumatraPDF generic printing methods are not very well suited compared to a Dedicated Barcode Software solution.  
Actually unless using Zebra PDF Direct or an equivalent SDK then PDF is the absolute worst format for Labels on Thermal Printers.

Anyway the issues will be down to scale and layout settings which depend on your target media so if the labels are same size as input then printout needs to be unscaled thus use `noscale` for scaling on command line AND in settings
```
PrinterDefaults [
	PrintScale = none
]
```

At that point the only problems should be rotation and paperkind
Rotation has a new setting to not autorotate thus we are left with Paperkind and that's the biggest problem, that Windows insists on a default of A4 for metric or Letter for Imperial so those need to be overridden in the default printer OR specified from the Forms that Windows keeps track of.

**Reply by dominiosc** [reply]
You are correct in everything you indicate,
blurry, wrong size and doesn't take into account cropped margins.
I use driver sdk of TSC printer, the format comes thus from the transport agency.

This is only one page of third case, it happens throughout the all pages of brochure.
https://www.dropbox.com/t/Cgv6l5ujS11g92Wf

**Reply by GitHubRulesOK** [reply]
the issue is not in official 3.5.2 only in current pre-release so you should for stable printing be using 3.5.2 then if there is a problem such as you pointed out (Thank You) it can probably be fixed for official 3.6

**Comment by GitHubRulesOK** [comment]
@kjk there is a problem with the way RECENT SumatraPDF is NOT respecting/using CropBox for PDF display.

From samples above the page is shown as A4 (Media) but it should only render smaller CropBox

K0TZK2MS-nodata.pdf
2 0 obj  
<</Resources 9 0 R /MediaBox[ 0 0 842 595]/Contents 10 0 R /Type/Page/Parent 1 0 R /CropBox[ 20 295 420 585]>>

MR73BEBP-nodata.pdf (/MediaBox[ 0 0 595 842] /CropBox[ 55 427 395 777])

16 0 obj  
<</Resources<</XObject<</I0 5 0 R >>/ExtGState<</FXE1 32 0 R >>/ProcSet[/PDF/Text/ImageB/ImageC/ImageI]/Font<</F0 8 0 R /F1 10 0 R /F2 12 0 R /F3 14 0 R >>>>/MediaBox[ 0 0 595 842]/Parent 1 0 R /Contents 4 0 R /Type/Page/CropBox[ 55 427 395 777]/Group<</CS/DeviceRGB/S/Transparency/I false/K false>>>>

**Comment by dominiosc** [comment]
In 15.888 is fixed. Thanks
